Fixes #531: Unable to deploy without branding settings feature

tpires opened the pull request:
Signed-off-by: Tiago Pires <tandrepires@gmail.com>

## ✏️ Changes

As stated in https://github.com/auth0/auth0-deploy-cli/issues/531 , this PR fixes the deploy for auth0 tenants that don't have the branding settings feature enabled. It also allows to deploy only page templates, or only branding settings.

## 🔗 References

Original GitHub issue: https://github.com/auth0/auth0-deploy-cli/issues/531

## 🎯 Testing

Added unit tests for the Auth0 Branding directory handler.

willvedd commented on test/context/directory/attackProtection.test.js:
```diff
@@ -9,10 +9,13 @@ describe('#directory context attack-protection', () => {
-        'brute-force-protection.json': '{"enabled": "@@BRUTE_FORCE_PROT_ENABLED@@", "shields": ["block", "user_notification"], "mode": "count_per_identifier_and_ip", "allowlist": [], "max_attempts": 10}',
-        'suspicious-ip-throttling.json': '{"enabled": true, "shields": ["block", "admin_notification"], "allowlist": ["127.0.0.1"], "stage": {"pre-login": {"max_attempts": 100, "rate": 864000}, "pre-user-registration": {"max_attempts": 50, "rate": 1200}}}'
-      }
+        'breached-password-detection.json':
```
**Thought:** I wonder how this file got checked-in without formatting. Since it's just a single file, I'm ok with it being here.

willvedd commented on src/context/directory/handlers/branding.ts:
```diff
@@ -10,17 +10,24 @@ type ParsedBranding = ParsedAsset<'branding', Asset>;
   );
 
-  if (!existsMustBeDir(brandingTemplatesFolder)) return { branding: null };
+  if (!existsMustBeDir(brandingTemplatesFolder)) return { branding: branding };
```
**Nitpick:**
```suggestion
  if (!existsMustBeDir(brandingTemplatesFolder)) return { branding };
```

tpires replied:
by modifying the variable above to `brandingSettings` this has to be `branding: brandingSettings`.

willvedd commented on test/context/directory/branding.test.js:
```diff
@@ -50,6 +50,51 @@ describe('#directory context branding', () => {
+    const context = new Context(config, mockMgmtClient());
+    await context.load();
+
+    expect(context.assets.branding).to.be.an('object');
```
**Suggestion:** Can probably remove this assertion as it's enforced with the deep equal check below.

willvedd commented on test/context/directory/branding.test.js:
```diff
@@ -50,6 +50,51 @@ describe('#directory context branding', () => {
+    await context.load();
+
+    expect(context.assets.branding).to.be.an('object');
+    expect(context.assets.branding).to.not.have.property('templates');
```
**Suggestion:** Can we add a deep equal assertion here for the expected branding object? Something like: 

```
expect(context.assets.branding).to.deep.equal(brandingSettings)
```

Would really strengthen this test IMO.

willvedd commented on src/context/directory/handlers/branding.ts:
```diff
@@ -10,17 +10,24 @@ type ParsedBranding = ParsedAsset<'branding', Asset>;
   if (!existsMustBeDir(brandingDirectory)) return { branding: null };
 
-  const branding = loadJSON(path.join(brandingDirectory, 'branding.json'), context.mappings);
+  const branding = (() => {
```
**Idea:** What do you think about renaming this `brandingSettings`? That way it's very clear what this object contains. Would require a couple other lines to change, but I think it would solve the original confusion as to why this bug was introduced in the first place.

tpires replied:
makes sense and it clears up any confusion around the naming.

willvedd reviewed (changes requested):
Thanks for taking the time to submit this! Overall, this contribution is pretty solid. You've very clearly identified and addressed a bug here. Just a few suggestions and requests to make the tests a bit stronger but don't see why this wouldn't get approved.

tpires commented:
@willvedd thanks for the quick reply. addressed your comments. circleci seems to be failing due to `missing npm command` - probably a restart failed steps should fix it.

sergiughf commented:
CircleCI was failing due to a bug introduced in the current version. Fix is https://github.com/CircleCI-Public/cimg-node/pull/238 but not yet merged. We pinned version 18.0.0 as the "current" version for now to make the CI tests pass.

sergiughf commented:
Thanks again for the contribution @tpires !
